.grit/patterns/ES6Exports.md

```diff
@@ -16,13 +16,23 @@ or {
             $vals <: some let($key, $name, $val, $match, $prop) ObjectProperty(key=$key, value=$name) as $prop => . where {
                 $name <: Identifier()
                 $exportedVals = [... $exportedVals, $prop]
-                $program <: contains or {
-                    `const $name = $val`
-                    `let $name = $val`
-                    `var $name = $val`
-                    `const $name = $val`
-                    FunctionDeclaration(id=$name)
-                } as $match => ExportNamedDeclaration(declaration=$match)
+                or {
+                    // special case of exporting a require() - see ES6Import pattern
+                    $program <: contains or {
```

Comment:
I don't think this i the level to split things at - why not just incorporate the `require` examples into the `or` clause below?

Reply:
that `or` clause rewrites the entire thing into `ExportNamedDeclaration(declaration=$match)`, didn't see a clean alternative other than this which seems fine to read

Reply:
The problem with this is it will traverse the whole program *twice*.

Reply:
Here you can combine the two:
```
                    $program <: contains or {
                        // special case of exporting a require() - see ES6Import pattern
                        or {
                            // TODO difficult to handle matching a sublist of the module.exports
                            // `const { $imports } = require($val)` => `export { $transformed } from "$val"` where {
                            //     $imports <: some bubble($transformed, $name) {ObjectProperty(key=$subkey, value=$subval) where {
                            //         $subval <: $name
                            //         $transformed = [...$transformed, `$subkey as $subval`]
                            //     } }
                            // }
                            `const $name = require($val).default` => `export { default as $name } from "$val"`
                            `const $name = require($val).$foo` => `export { $foo as $name } from "$val"`
                            `const $name = require($val)` => `export { default as $name } from "$val"`
                        },
                        or {
                            `const $name = $val`
                            `let $name = $val`
                            `var $name = $val`
                            `const $name = $val`
                            FunctionDeclaration(id=$name)
                        } as $match => ExportNamedDeclaration(declaration=$match)
                    }
                    ```

Reply:
ah, i see, it's the internal `contains`. thanks, updated!